<?xml version="1.0" ?>

<launch>
    <arg name="robot_type" default="$(env ROBOT_TYPE)" doc="Robot type: [a1, aliengo, go1, laikago]"/>

    <param name="urdfFile" value="/tmp/legged_control/$(arg robot_type).urdf"/>
    <param name="taskFile" value="$(find legged_perceptive_controllers)/config/$(arg robot_type)/task.info"/>
    <param name="referenceFile" value="$(find legged_controllers)/config/$(arg robot_type)/reference.info"/>
    <param name="gaitCommandFile" value="$(find legged_controllers)/config/$(arg robot_type)/gait.info"/>

    <rosparam file="$(find legged_perceptive_controllers)/config/default.yaml" command="load"/>

    <node name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/perceptive_controller
"/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command"
          output="screen"/>

    <node pkg="legged_controllers" type="legged_target_trajectories_publisher" name="legged_robot_target"
          output="screen"/>

    <include file="$(find legged_perceptive_controllers)/launch/elevation_mapping.launch">
    </include>
</launch>
